app/models/partner.rb

```diff
@@ -101,18 +102,21 @@ def approvable?
   # better to extract this outside of the model
   def self.import_csv(csv, organization_id)
     errors = []
+    warnings = []
     organization = Organization.find(organization_id)
 
     csv.each do |row|
       hash_rows = Hash[row.to_hash.map { |k, v| [k.downcase, v] }]
 
       svc = PartnerCreateService.new(organization: organization, partner_attrs: hash_rows)
       svc.call
-      if svc.errors.present?
+      if svc.errors.present? && svc.partner.errors.blank?
+        warnings << "#{svc.partner.name}: #{svc.errors.full_messages.to_sentence}"
+      elsif svc.errors.present?
```

Comment:
@cielf I wanted to highlight this logic to you.  

If a user attempts to upload a partner with an email and name that is already in use, in addition to a misspelled storage location name, both the email/name messages and the storage location message will display as errors.  If you want, I could look into displaying the email/name error messages as errors and the storage location message as a warning.  Let me know what you prefer.

Reply:
My initial take is that if the partner is not getting added, having everything as errors won't be a big deal.